### PR TITLE
Meaningful error message when it is not possible to summon subclass schema.

### PR DESCRIPTION
**Pull request by rafalambrozewicz**
In `generateOneOfWrapped` method, added a meaningful error message for a case when it is not possible to summon subclass schema.

### CODE REVIEW

File: core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala [hunk 1]
```diff
@@ -252,9 +252,13 @@ private[tapir] object SchemaCompanionMacros {
 
       val subclassesSchemas: List[Expr[(String, Schema[_])]] = subclasses.map(subclass =>
         TypeIdent(subclass).tpe.asType match {
-          case '[f] =>
-            val subSchema = Expr.summon[Schema[f]].get
-            '{ ${ Expr(subclass.name) } -> Schema.wrapWithSingleFieldProduct($subSchema)($conf) }
+          case t @ '[f] =>
+            val opSubSchema = Expr.summon[Schema[f]]
+            if (opSubSchema.isEmpty) {
```
Comment (adamw) [inline core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:257]: nicer:

```
Expr.summon[Schema[f]] match
  case None => report.errorAndAbort(...)
  case Some(...) => ...
```

Comment (adamw) [reply core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala:257]: (ot only nicer but in general less error prone, as you can't get it wrong later - generally using `.get` is discouraged

Comment (rafalambrozewicz) [reply core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala]: Great suggestion. Applied. Thanks!

**Comment by adamw** [comment]
Thanks :)

**Pull request merged**